worker/games.py

```diff
@@ -326,41 +332,47 @@ def run_game(p, remote, result, spsa, spsa_tuning, tc_limit):
         spsa['losses'] = wld[1]
         spsa['draws'] = wld[2]
 
-      update_succeeded=False
-      for _ in range(0,5):
-        try:
-          t0 = datetime.datetime.utcnow()
-          req = requests.post(remote + '/api/update_task', data=json.dumps(result), headers={'Content-type': 'application/json'}, timeout=HTTP_TIMEOUT).json()
-          print("Task updated successfully in %ss" % ((datetime.datetime.utcnow() - t0).total_seconds()))
-          if not req['task_alive']:
-            # This task is no longer neccesary
-            print('Server told us task is no longer needed')
+      num_games_finished = sum(wld)
+      if (num_games_finished >= 2 and num_games_finished % concurrency == 0) or num_games_finished == games_to_play:
+        # Send game results after 1 game per worker CPU is played or after the last game is played
+
+        update_succeeded = False
+        # Attempt to send game results to the server. Retry a few times upon error
+        for _ in range(0, 5):
+          try:
+            t0 = datetime.datetime.utcnow()
+            response = send_api_post_request(remote + '/api/update_task', result).json()
+            print("  Task updated successfully in %ss" % ((datetime.datetime.utcnow() - t0).total_seconds()))
+            if not response['task_alive']:
+              # This task is no longer neccesary
+              print('Server told us task is no longer needed')
+              kill_process(p)
+              return response
+            update_succeeded = True
+            break
+          except Exception as e:
+            sys.stderr.write('Exception from calling update_task:\n')
+            print(e)
+            # traceback.print_exc(file=sys.stderr)
+          time.sleep(HTTP_TIMEOUT)
+
+          if not update_succeeded:
```

Comment:
If I'm reading right this `if` seems to be in the `for` cycle, in this way at the first exception the process will be terminated.

Reply:
No it will print the exception and continue the loop at the sleep...

Reply:
After the sleep it should execute `if not update_succeeded:` and kill the process.
In master `if not update_succeeded:` is out that `for` cycle.

Reply:
Hmm in my version of games.py ``` if not update_succeeded ``` is outside the loop.

Reply:
Ah, yes it should be outside the loop...

Reply:
oops, fixed